### ISSUE TITLE
Only depend on required library versions

### ISSUE THREAD

**Issue by hpoul** [Daegalus/dart-otp#41]
Hello,
the recent update enforces `crypto` to at least `3.0.2`. I assume that the actual requirement could be `^3.0.0` as there are probably no changes in 3.0.1 or 3.0.2 which are relevant to otp?
by having a less strict dependency it would allow users to update one dependency at a time.. right now I can't update because `flutter_driver` depends on crypto `3.0.1`:

```
Because every version of flutter_driver from sdk depends on crypto 3.0.1 and otp >=3.1.0 depends on crypto ^3.0.2, flutter_driver from sdk is incompatible with otp >=3.1.0.
So, because authpass depends on both otp ^3.1.0 and flutter_driver from sdk, version solving failed.
pub get failed (1; So, because authpass depends on both otp ^3.1.0 and flutter_driver from sdk, version solving failed.)
```

thanks,
  Herbert

**Comment by Daegalus** [comment]
You are correct. I have a busy day today, but I will make the changes tonight after the baby goes to bed. 

I apologies for the inconvenience.

**Comment by hpoul** [comment]
No worries, not that urgent ;-) Would be nice to keep in mind for future releases.
thanks

**Comment by Daegalus** [comment]
Just published 3.1.1, hopefully that fixes it. and I will make sure to consider such changes in the future and how they might interact with other libraries people use. Thanks for pointing this out.
